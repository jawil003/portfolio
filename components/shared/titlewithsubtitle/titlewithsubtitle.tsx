import { css } from "@emotion/react";
import React from "react";
import { FlexContainer } from "../flexcontainer/flexcontainer";
import { Typography } from "../typography";

export interface Props {
  heading: string;
  description: string;
  className?: string;
}

/**
 * An HeaderWithSubtitle React Component.
 * @author
 * @version 0.1
 */
export const TitleWithSubtitle: React.FC<Props> = ({
  heading,
  description,
  children,
  className,
}) => {
  return (
    <FlexContainer
      rowGap="1em"
      className={className}
      css={[
        css`
          & {
            display: inline-block;
            margin: 0;
            max-width: 500px;
          }
          & > .subtitle {
            width: 0;
            min-width: 100%;
          }
        `,
      ]}
    >
      <Typography
        bold="semi-bold"
        variant="h3"
        component="h1"
      >
        {heading}
      </Typography>
      <Typography
        className="subtitle"
        italic
      >
        {description}
      </Typography>
      {children}
    </FlexContainer>
  );
};

TitleWithSubtitle.displayName =
  "TitleWithSubtitle";
