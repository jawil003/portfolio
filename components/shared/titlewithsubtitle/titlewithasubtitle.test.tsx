import { render } from "@testing-library/react";
import React from "react";
import { TitleWithSubtitle } from "./titlewithsubtitle";

describe("TitleWithSubtitle", () => {
  test("Create TitleWithSubtitle", () => {
    const testTitleWithSubtitle = render(
      <TitleWithSubtitle
        heading="Test heading"
        description="test description"
      />,
    );

    expect(
      testTitleWithSubtitle.container
        .firstChild,
    ).toMatchSnapshot();
  });
});
