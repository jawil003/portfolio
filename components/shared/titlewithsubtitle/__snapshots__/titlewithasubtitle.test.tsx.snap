// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleWithSubtitle Create TitleWithSubtitle 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-flex-wrap: no-wrap;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  margin: calc(1em / -2) calc(0px / -2);
}

.emotion-0>* {
  margin: calc(1em / 2) calc(0px / 2);
}

.emotion-0 {
  display: inline-block;
  margin: 0;
  max-width: 500px;
}

.emotion-0>.subtitle {
  width: 0;
  min-width: 100%;
}

.emotion-1 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  letter-spacing: normal;
  word-spacing: normal;
  color: #000;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-1 {
  font-weight: 600;
  font-size: 3em;
}

.emotion-3 {
  text-align: start;
  font-style: italic;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  letter-spacing: normal;
  word-spacing: normal;
  color: #000;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-3 {
  font-weight: 400;
  font-size: 1em;
}

<div
  class="emotion-0"
>
  <h1
    class="emotion-1 emotion-2"
    color="#000"
    id="heading3"
  >
    Test heading
  </h1>
  <p
    class="subtitle emotion-3 emotion-4"
    color="#000"
    id="body1"
  >
    test description
  </p>
</div>
`;
